### ISSUE TITLE
Add CONTRIBUTING.md

### ISSUE THREAD

**Issue by markbandstra** [lbl-anp/becquerel#121]
Many repos I have seen have guides for contributions, e.g.:

https://github.com/numpy/numpy/blob/master/CONTRIBUTING.md
https://github.com/numba/numba/blob/master/CONTRIBUTING.md
https://github.com/github/gitignore/blob/master/CONTRIBUTING.md

I think we might want to move the Code Style Guide and Linter sections out of our README and into such a file. It might also be helpful to add a checklist to describe the workflow, e.g.:

- [ ] Branch off of `develop` and name the branch `feature-XX` or `issue-XX`
- [ ] Develop your feature or fix
- [ ] Write tests to cover all use cases
- [ ] Ensure test coverage is >95%
- [ ] Ensure all tests pass (`python setup.py test`)
- [ ] Ensure there are no linter issues and style is consistent
- [ ] Push branch to GitHub
- [ ] Create a pull request

**Comment by jccurtis** [comment]
Additionally we could add linting to the CI build which will reduce manual checking of style enforcement, check [this](https://jeffknupp.com/blog/2016/12/09/how-python-linters-will-save-your-large-python-project/) out.

**Comment by markbandstra** [comment]
Good point, I would be in favor of that. We should probably modify our pylintrc to remove the most annoying and unnecessary warnings though.

**Comment by jccurtis** [comment]
Definitely, lets chat about how that will look next Tuesday as well. 😉
